lib/elixir/lib/calendar.ex

```diff
@@ -789,29 +974,96 @@ defmodule Time do
   """
   @spec compare(Calendar.time, Calendar.time) :: :lt | :eq | :gt
   def compare(time1, time2) do
-    case {to_tuple(time1), to_tuple(time2)} do
+    {parts1, ppd1} = to_day_fraction(time1)
+    {parts2, ppd2} = to_day_fraction(time2)
+
+    case {parts1 * ppd2, parts2 * ppd1} do
       {first, second} when first > second -> :gt
       {first, second} when first < second -> :lt
       _ -> :eq
     end
   end
 
+  @doc """
+  Converts the Time struct to a different calendar.
+  """
+  @spec convert(Time.t, Calendar.calendar) :: Time.t
+  # No conversion required if already of the target calendar.
+  def convert(%{calendar: calendar, hour: _, minute: _, second: _, microsecond: _} = time, calendar) do
+    {:ok, time}
+  end
+
+  def convert(%{calendar: calendar, hour: _, minute: _, second: _, microsecond: _} = time, calendar) do
+    result_time =
+      time
+      |> to_day_fraction
+      |> calendar.time_from_day_fraction
+    {:ok, result_time}
+  end
+
+  @spec convert!(Time.t, Calendar.calendar) :: Time.t
+  def convert!(time, calendar) do
+    case convert(time, calendar) do
+      {:ok, value} ->
+        value
+      {:error, reason} ->
+        raise ArgumentError, "cannot convert #{inspect time} to target calendar #{inspect calendar}, reason: #{inspect reason}"
+    end
+  end
+
+  @doc """
+  Returns a day_fraction that represents the difference
+  between two Time Structs.
+  """
+  @spec diff(Time.t, Time.t) :: Calendar.rata_die
+  def diff(%Time{} = time1, %Time{} = time2) do
```

Comment:
As per our discussion, this should returns Calendar.ISO seconds. Also the spec right now is wrong, it doesn't return rata_die, rather, day_fraction. But it doesn't matter, it should be seconds, microseconds, etc as in the NaiveDateTime implementation.